### ISSUE TITLE
Parse error with regex (dbt)

### ISSUE THREAD

**Issue by daniel-murray** [sqlfluff/sqlfluff#842]
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

I'm getting an unexpected parsing error.

Here's a snippet of the code. It's a safe case from string to float.
```sql
{%- macro number_to_float(column_name) -%}
case
    when {{ column_name }} ~ '^([0-9]){1,}(\.)([0-9]{1,})$' then {{ column_name }}::float
    when {{ column_name }} ~ '^[0-9]{1,}$' then {{ column_name }}::float
    else null
end
{%- endmacro -%}
```

## Expected Behaviour
No passing error.

## Observed Behaviour
```
L:  18 | P:   9 |  PRS | Found unparsable section: "case\n when value ~
                       | '^([0-9]){1,}(\\.)(..."
```

## Steps to Reproduce
Add this macro to a column in any dbt model using `{{ number_to_float("value") }}`, or the raw sql (error is not limited to it's use in a macro.)

## Version
sqlfluff, version 0.4.1
Python 3.7.5

## Configuration
```
[sqlfluff]
dialect=postgres
templater=dbt
exclude_rules = L009

[sqlfluff:indentation]
template_blocks_indent = False
```

**Comment by barrywhart** [comment]
Could this be related to #828?

**Comment by alanmcruickshank** [comment]
I thought this was the regex strings, but I think it's just that we don't have the `~` operator set up properly. PR is live. Thanks for the great bug report @daniel-murray  !

**Comment by alanmcruickshank** [comment]
This was fixed with #932
